### PR TITLE
RELATED: RAIL-4066 measures backend spi

### PR DESCRIPTION
**Pull request by plavemar**
- add getMeasures() into workspace measure service to be able to acquire limited measure objects only

JIRA: RAIL-4066

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7375/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9231/ : FAILURE in 2m 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7376/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9232/ : FAILURE in 2m 39s

File: libs/sdk-backend-tiger/src/backend/workspace/measures/index.ts [hunk 1]
```diff
@@ -201,4 +202,20 @@ export class TigerWorkspaceMeasures implements IWorkspaceMeasuresService {
...
+
+            return allMetrics.filter((metric) =>
+                measureRefs.find((metricRef) => areObjRefsEqual(metricRef, metric.ref)),
+            );
+        });
+    };
+}
+
+function loadMetrics(client: ITigerClient, workspaceId: string): Promise<IMeasureMetadataObject[]> {
+    return MetadataUtilities.getAllPagesOf(client, client.entities.getAllEntitiesMetrics, { workspaceId })
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/backend/workspace/measures/index.ts:218]: Please try using the RSQL to filter on the backend if the requested number of metrics is small enough to fit the query string.

File: libs/sdk-backend-spi/src/workspace/measures/index.ts [hunk 2]
```diff
@@ -52,4 +52,12 @@ export interface IWorkspaceMeasuresService {
      * @returns promise of references
      */
     getMeasureReferencingObjects(measureRef: ObjRef): Promise<IMeasureReferencing>;
+
+    /**
+     * Get all metadata object for given measure references.
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/measures/index.ts:57]: Typo: objects

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7387/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9237/ : SUCCESS in 16m 29s

File: libs/sdk-backend-spi/src/workspace/measures/index.ts [hunk 3]
```diff
@@ -52,4 +52,19 @@ export interface IWorkspaceMeasuresService {
      * @returns promise of references
      */
     getMeasureReferencingObjects(measureRef: ObjRef): Promise<IMeasureReferencing>;
+
+    /**
+     * Get all metadata objects for given measure references.
+     *
+     * @remarks
+     * If the array of the given references is too large, the function must load all the measures
```
Comment (kandl) [inline libs/sdk-backend-spi/src/workspace/measures/index.ts:60]: I think we should hide this in the implementation + I think it's not necessary. If the array of the given references is too large, we can divide it to multiple requests, run them in parallel and then return just all the results.

**Review by no23reason — APPROVED** [review]
LGTM

File: libs/sdk-backend-spi/src/workspace/measures/index.ts (hunk 3 above)
Comment (no23reason) [reply libs/sdk-backend-spi/src/workspace/measures/index.ts:60]: Discussed offline, we decided that this further optimization would not likely help in common use cases.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2380

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2135/ : SUCCESS in 1m 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2420/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1241/ : SUCCESS in 15m 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9238/ : SUCCESS in 16m 25s

**Pull request merged**